test_communication/CMakeLists.txt

```diff
@@ -66,20 +66,26 @@ if(BUILD_TESTING)
       # adding test for each message type
       foreach(message_file ${message_files})
         get_filename_component(TEST_MESSAGE_TYPE "${message_file}" NAME_WE)
+
+        set(SKIP_TEST "")
         # TODO(dirk-thomas) publishing Bounded/DynamicArrayPrimitives in OpenSplice is buggy
         # https://github.com/ros2/rclcpp/issues/219
-        if((NOT TEST_MESSAGE_TYPE STREQUAL "BoundedArrayPrimitives" AND NOT TEST_MESSAGE_TYPE STREQUAL "DynamicArrayPrimitives") OR NOT rmw_implementation STREQUAL "rmw_opensplice_cpp")
-          ament_add_test(
-            "${target}${target_suffix}__${TEST_MESSAGE_TYPE}"
-            COMMAND "$<TARGET_FILE:${target}${target_suffix}>" "${TEST_MESSAGE_TYPE}"
-            TIMEOUT 15
-            GENERATE_RESULT_FOR_RETURN_CODE_ZERO
-            ENV RCL_ASSERT_RMW_ID_MATCHES=${rmw_implementation})
-          set_tests_properties(
-            "${target}${target_suffix}__${TEST_MESSAGE_TYPE}"
-            PROPERTIES REQUIRED_FILES "$<TARGET_FILE:${target}${target_suffix}>"
-          )
+        if(
+          (TEST_MESSAGE_TYPE STREQUAL "BoundedArrayPrimitives" OR TEST_MESSAGE_TYPE STREQUAL "DynamicArrayPrimitives") AND
+          rmw_implementation1 STREQUAL "rmw_opensplice_cpp"
+        )
+          set(SKIP_TEST "True")
```

Comment:
It is working for nose test only indeed, thanks for catching it. Fixed in https://github.com/ros2/system_tests/pull/175/commits/8938b1f1a89c7156aadcd0a0a31eae8b200d70ca
Current CI doesn't test opensplice so it wouldn't have appeared in the CI test results